crates/cli/src/main.rs

```diff
@@ -92,25 +93,32 @@ async fn main() -> Result<()> {
             };
         }
         Some(Commands::Sim { config: _ }) => {
-            // Provider we will use.
-            let client = get_provider().await;
+            // Create a `ExecutionManager` where we can run simulations.
+            let mut manager = ExecutionManager::new();
 
-            // This is the only part of main that uses a provider/client. The client doesn't actually do anything, but it is a necessary inner for ContractDeployer
-            let contract_deployer = bindings::hello_world::HelloWorld::deploy(client, ()).unwrap();
-            let initialization_bytes = contract_deployer.deployer.tx.data().unwrap();
+            // Get a BaseContract for the ERC-20 contract from the ABI.
+            let erc20_contract = BaseContract::from(bindings::erc20::ERC20_ABI.clone());
 
-            let hello_world_contract =
-                BaseContract::from(bindings::hello_world::HELLOWORLD_ABI.clone());
+            // Choose name and symbol for the constructor args required by ERC-20 contracts.
+            let name = "ArbiterToken";
```

Comment:
We can address this in a later PR, but if it's going to be too challenging to do the general deployer abstraction we can at least do it for `deploy_token` or something.

Reply:
We can certainly do a general deploy abstraction! I'd enjoy making that actually.